src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/FontCache/FontCacheUtil.cs

```diff
@@ -661,7 +661,7 @@ private static string NormalizeFontFamilyReference(string fontFamilyReference, i
             {
                 // No fragment separator. The entire string is a family name so convert to uppercase
                 // and add a fragment separator at the beginning.
-                return "#" + fontFamilyReference.Substring(startIndex, length).ToUpperInvariant();
+                return string.Concat("#", fontFamilyReference.AsSpan(startIndex, length)).ToUpperInvariant();
```

Comment:
Could you call .ToUpperInvariant() on the span instead of on the string ?

It would save an allocation of string from .ToUpperInvariant(). I don't think we need to call .ToUpperInvariant() on "#", since it's a symbol.

Reply:
There isn't a `ToUpperInvariant` that produces a string from a span.  There is one that operates in-place, but that requires a `Span<char>` not a `ReadOnlySpan<char>`.

The only downside here is, if upper-casing is required, we allocate two additional bytes unnecessarily.  In contrast, the original would end up allocating an entire additional string.

Reply:
Sorry, I checked the API doc a bit to quick. I didn't notice that it required a writable span as an argument. Thanks